simplyblock_core/shell_utils.py

```diff
@@ -3,7 +3,10 @@
 
 
 def run_command(cmd):
-    process = subprocess.Popen(
-        cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    stdout, stderr = process.communicate()
-    return stdout.strip().decode("utf-8"), stderr.strip(), process.returncode
+    try:
+        process = subprocess.Popen(
+            cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        return stdout.strip().decode("utf-8"), stderr.strip(), process.returncode
+    except Exception as e:
```

Comment:
Would it make sense to let the caller handle this exception instead?

Reply:
Definitely, though we make little use of exception based error handling yet. It would make sense to change set, it simplifies lots of cumbersome error handling which is easy to get wrong. The `subprocess` module already has helpers that would be suitable for this, something like this would be what I'd prefer:
```
def run_command(command):
    return subprocess.check_output(command.split(' '), stderr=subprocess.PIPE, text=True)
```
It'd throw automatically including a descriptive exception, including the stderr output. We'd have to change all call-sites to handle this appropriately though. I didn't make that change since I thought avoiding exceptions was a conscious descision, if we want to use more of them I can adapt the PR.